UNIXFSv1.md

```diff
@@ -0,0 +1,487 @@
+		- [Metadata in the file](#metadata-in-the-file)
+		- [Side trees](#side-trees)
+		- [Side database](#side-database)
+
+## Implementations
+
+- JavaScript
+  - Data Formats - [unixfs](https://github.com/ipfs/js-ipfs-unixfs)
+  - Importer - [unixfs-importer](https://github.com/ipfs/js-ipfs-unixfs-importer)
+  - Exporter - [unixfs-exporter](https://github.com/ipfs/js-ipfs-unixfs-exporter)
+- Go
+  - [`ipfs/go-ipfs/unixfs`](https://github.com/ipfs/go-ipfs/tree/b3faaad1310bcc32dc3dd24e1919e9edf51edba8/unixfs)
+  - Protocol Buffer Definitions - [`ipfs/go-ipfs/unixfs/pb`](https://github.com/ipfs/go-ipfs/blob/b3faaad1310bcc32dc3dd24e1919e9edf51edba8/unixfs/pb/unixfs.proto)
+
+## Data Format
+
+The UnixfsV1 data format is represented by this protobuf:
+
+```protobuf
+message Data {
+	enum DataType {
+		Raw = 0;
+		Directory = 1;
+		File = 2;
+		Metadata = 3;
+		Symlink = 4;
+		HAMTShard = 5;
+	}
+
+	required DataType Type = 1;
+	optional bytes Data = 2;
+	optional uint64 filesize = 3;
+	repeated uint64 blocksizes = 4;
+	optional uint64 hashType = 5;
+	optional uint64 fanout = 6;
+	optional uint32 mode = 7;
+	optional UnixTime mtime = 8;
+}
+
+message Metadata {
+	optional string MimeType = 1;
+}
+
+message UnixTime {
+	required int64 Seconds = 1;
+	optional fixed32 FractionalNanoseconds = 2;
+}
+```
+
+### IPLD `dag-pb`
+
+A very important other spec for unixfs is the [`dag-pb`](https://ipld.io/specs/codecs/dag-pb/spec/) IPLD spec:
+
+```protobuf
+message PBLink {
+  // binary CID (with no multibase prefix) of the target object
+  optional bytes Hash = 1;
+
+  // UTF-8 string name
+  optional string Name = 2;
+
+  // cumulative size of target object
+  optional uint64 Tsize = 3; // also known as dagsize
+}
+
+message PBNode {
+  // refs to other objects
+  repeated PBLink Links = 2;
+
+  // opaque user data
+  optional bytes Data = 1;
+}
+```
+
+The two different schemas plays together and it is important to understand their different effect,
+- `dag-pb` also named `PBNode` is the "outside" protobuf message, it is the first one you decode. It contain the list of links and some "opaque user data".
+- The `Data` message is the "inside" protobuf message, this can be decoded by first decoding the `PBNode` object and then decoding `Data` from the bytes inside `PBNode.Data` field, this will contain all the rest of information.
+
+This mean we deal with protobuf inside protobuf. This is for history reasons.
+This spec sometimes show this a one object for clarity reasons, when that happen it is implied that the `PBNode.Data` field is encoded into prototbuf.
+
+### Glossary
+
+- Node, Block
+  A node is a word from graph theory, this is the smallest unit present in the graph.
+  Due to how unixfs work, there is a 1 to 1 mapping between nodes and blocks.
+- File
+  A file is some container over an arbitrary sized amounts of bytes.
+  Files can be said to be single block, or multi block, in the later case they are the concatenation of multiple children files.
```

Comment:
> Files can be said to be single block, or multi block, in the later case they are the concatenation of data stored in children blocks (as `Data` field of `dag-pb` or a `raw` leave blocks without protobuf envelope).

My understanding of this sentence is that a multi block file only concatenates the `PBNode.Data.Data`, while this is wrong, it actually concatenate the content of the file whatever it may be (for a dag-pb file that `PBNode.Data.Data` + the concatenation of the links's content (in a DFS maner)).